cluster/router/tag/match.go

```diff
@@ -99,22 +99,39 @@ func requestTag(invokers []protocol.Invoker, url *common.URL, invocation protoco
 	var (
 		addresses []string
 		result    []protocol.Invoker
+		match     []common.ParamMatch
 	)
 	for _, tagCfg := range cfg.Tags {
 		if tagCfg.Name == tag {
 			addresses = tagCfg.Addresses
+			match = tagCfg.Match
 		}
 	}
-	if len(addresses) == 0 {
-		// filter tag does not match
-		result = filterInvokers(invokers, tag, func(invoker protocol.Invoker, tag interface{}) bool {
-			return invoker.GetURL().GetParam(constant.Tagkey, "") != tag
+
+	// only one of 'match' and 'addresses' will take effect if both are specified.
+	if len(match) != 0 {
+		result = filterInvokers(invokers, match, func(invoker protocol.Invoker, match interface{}) bool {
+			matches := match.([]common.ParamMatch)
+			isMatch := true
+			for _, m := range matches {
+				if !m.IsMatch(invoker.GetURL()) {
+					isMatch = false
+				}
+			}
+			return !isMatch
```

Comment:
`isMatch` is an unnecessary variable here:

```suggestion
			matches := match.([]common.ParamMatch)
			for _, m := range matches {
				if !m.IsMatch(invoker.GetURL()) {
					return true
				}
			}
			return false
```